central/cve/image/datastore/internal/search/searcher_impl.go

```diff
@@ -0,0 +1,91 @@
+package search
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/cve/image/datastore/internal/store/postgres"
+	"github.com/stackrox/rox/central/cve/index"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/search"
+)
+
+type searcherImpl struct {
+	storage  postgres.Store
+	indexer  index.Indexer
+	searcher search.Searcher
+}
+
+func (ds *searcherImpl) SearchCVEs(ctx context.Context, q *v1.Query) ([]*v1.SearchResult, error) {
+	results, err := ds.getSearchResults(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	return ds.resultsToSearchResults(ctx, results)
+}
+
+func (ds *searcherImpl) Search(ctx context.Context, q *v1.Query) ([]search.Result, error) {
+	return ds.getSearchResults(ctx, q)
```

Comment:
not sure why we need a function if it's a one liner anyways

Reply:
copy-pasted template used everywhere. :) I am sure we will have this replaced by searcher auto-gen.